--- a/setup.py
+++ b/setup.py
@@ -12,6 +12,8 @@
 import numpy as np
 
 
+is_conda = 'CONDA_PREFIX' in os.environ
+
 ''' C extensions
 
 CGAL Compilation
@@ -92,12 +94,11 @@
         os.path.join(tbb_path, 'tbb', 'lib', 'libtbb.dylib'),
         os.path.join(tbb_path, 'tbb', 'lib', 'libtbbmalloc.dylib'),
     ]
-else:
-    raise OSError('OS not supported!')
+if not is_conda:
+    print("Warning: Conda not detected. Attempting to use system paths.")
 
 
 #### Setup compilation arguments
-is_conda = 'CONDA_PREFIX' in os.environ
 
 if sys.platform == 'win32':
     petsc_libs = ['libpetsc', 'msmpi']
@@ -147,18 +148,25 @@
     petsc_extra_link_args = None
 
     cgal_libs = ['mpfr', 'gmp', 'z', 'tbb', 'tbbmalloc', 'pthread']
-    cgal_include = [
-        np.get_include(),
-        CGAL_headers,
-        eigen_headers,
-        tbb_headers,
-        'simnibs/external/include/linux/mpfr',
-        'simnibs/external/include/linux/gmp'
-    ]
-    # To find the boost headers if installed with conda
+    
     if is_conda:
-        cgal_include += [os.path.join(os.environ['CONDA_PREFIX'], 'include')]
-    cgal_dirs = ['simnibs/external/lib/linux']
+        cgal_include = [
+            np.get_include(),
+            os.path.join(os.environ['CONDA_PREFIX'], 'include'),
+            os.path.join(os.environ['CONDA_PREFIX'], 'include', 'eigen3'),
+        ]
+        cgal_dirs = [os.path.join(os.environ['CONDA_PREFIX'], 'lib'), 'simnibs/external/lib/linux']
+    else:
+        cgal_include = [
+            np.get_include(),
+            CGAL_headers,
+            eigen_headers,
+            tbb_headers,
+            'simnibs/external/include/linux/mpfr',
+            'simnibs/external/include/linux/gmp'
+        ]
+        cgal_dirs = ['simnibs/external/lib/linux']
+
     cgal_runtime = ['$ORIGIN/../../external/lib/linux']
     # Add -Os -flto for much smaller binaries
     cgal_compile_args = [
@@ -379,13 +387,15 @@
             )
         )
         if self.force or changed_meshing:
-            download_and_extract(CGAL_url)
+            if not is_conda:
+                download_and_extract(CGAL_url)
             if self.inplace:
                 build_lib = ""
             else:
                 build_lib = self.build_lib + "/"
 
-            install_lib(tbb_url, tbb_path, tbb_libs, build_lib)
+            if not is_conda:
+                install_lib(tbb_url, tbb_path, tbb_libs, build_lib)
 
         # Compile
         build_ext.run(self)
